armasec/token_security.py

```diff
@@ -153,3 +138,56 @@ async def __call__(self, request: Request) -> TokenPayload:
                 )
 
         return token_payload
+
+    def _load_all_managers(self) -> None:
+        for domain_config in self.domain_configs:
```

Comment:
The problem with this approach is that it will attempt to load all the managers on the first request it tries to authenticate. If you have 5 different domains, it will initialize all of them before it services the request.

I will need to think if there is a way we can optimize this, but for now it will be ok as-is.

Reply:
yeah, this logic isn't perfect indeed. However, 5 different domains a challenge... we can optimize the code during the time. For now, two different domains take about 4-5 seconds to load, and then the following requests take 10-15 miliseconds according to my experiments. I guess it's ok to proceed